homeassistant/components/prometheus/__init__.py

```diff
@@ -289,6 +294,19 @@ def _handle_climate(self, state):
             )
             metric.labels(**self._labels(state)).set(current_temp)
 
+        current_action = state.attributes.get(ATTR_HVAC_ACTION)
+        if current_action:
+            metric = self._metric(
+                "climate_action",
+                self.prometheus_cli.Gauge,
+                "Current HVAC action",
+                ["action"],
+            )
+            for action in CURRENT_HVAC_ACTIONS:
+                metric.labels(**dict(self._labels(state), action=action)).set(
+                    1 if action == current_action else 0
```

Comment:
Not sure if that's prettier, but it encapsulates the semantics pretty well IMO:
```suggestion
                    int(action == current_action)
```

Reply:
Done